### PR TITLE
don't sign X.509 certs

### PR DESCRIPTION
**Pull request by ripatel-fd**
#### Problem

- Addresses a TODO comment regarding a vulnerability 
- Removes the pkcs8 dependency 
- Removes logic to create X.509 signatures (X.509 signatures are ignored)
- Hardcodes the SAN (SAN is ignored)
- Removes the rcgen dependency

#### Summary of Changes

Fixes #

### CODE REVIEW

**Comment by lijunwangs** [comment]
Can you elaborate on what problems this PR solves? Why do we need this?

File: streamer/src/tls_certificates.rs [hunk 1]
```diff
@@ -1,58 +1,99 @@
 use {
-    pkcs8::{der::Document, AlgorithmIdentifier, ObjectIdentifier},
-    rcgen::{CertificateParams, DistinguishedName, DnType, RcgenError, SanType},
-    solana_sdk::{pubkey::Pubkey, signature::Keypair},
-    std::net::IpAddr,
+    solana_sdk::{pubkey::Pubkey, signature::Keypair, signer::Signer},
     x509_parser::{prelude::*, public_key::PublicKey},
 };
 
 pub fn new_self_signed_tls_certificate(
```
Comment (t-nelson) [inline streamer/src/tls_certificates.rs:6]: can we rename this method to acknowledge that it's producing a dummy cert?

**Review by t-nelson — COMMENTED** [review]
changes lgtm now. gonna tap in @lijunwangs and @behzadnouri for rationale

**Comment by behzadnouri** [comment]
* Can you please confirm that this change is backward compatible with current mainnet?
  i.e. if I spin up a staked node with this code on mainnet, other nodes still accept connections from my node and correctly recognize its pubkey?
* This is probably a dumb question, but regarding
  > The signature contained in the certificate is now deliberately invalid, as nodes don't verify it anyways.

  what exactly prevents pubkey spoofing with this code? i.e. a node claiming a pubkey which they don't possess the private key of.

**Comment by behzadnouri** [comment]
> Is there a simple CLI tool I can run that spins up a solana-streamer client or server? That would help with testing.
> Ideally the test setup is reproducible so you can confirm yourself as well.

I think the easiest would be to run a staked node on testnet and verify quic connections are ok, and other nodes extract the correct pubkey from the connection.

File: core/src/repair/quic_endpoint.rs [hunk 2]
```diff
@@ -123,11 +118,11 @@ pub(crate) fn new_quic_endpoint(
     runtime: &tokio::runtime::Handle,
     keypair: &Keypair,
     socket: UdpSocket,
-    address: IpAddr,
+    _address: IpAddr,
```
Comment (behzadnouri) [inline core/src/repair/quic_endpoint.rs:121]: should this be removed then?

File: turbine/src/quic_endpoint.rs [hunk 3]
```diff
@@ -96,7 +91,7 @@ pub fn new_quic_endpoint(
     runtime: &tokio::runtime::Handle,
     keypair: &Keypair,
     socket: UdpSocket,
-    address: IpAddr,
+    _address: IpAddr,
```
Comment (behzadnouri) [inline turbine/src/quic_endpoint.rs:94]: remove this arg?

**Review by behzadnouri — APPROVED** [review]

**Review by behzadnouri — DISMISSED** [review]

File: streamer/src/tls_certificates.rs [hunk 4]
```diff
@@ -1,58 +1,97 @@
...
-            parameters: None,
-        },
-        private_key: &private_key,
-        public_key: None,
-    };
-    let key_pkcs8_der = key_pkcs8
-        .to_der()
-        .expect("Failed to convert keypair to DER")
-        .to_der();
+    //
```
Comment (lijunwangs) [inline streamer/src/tls_certificates.rs:12]: Can you add a link to where the prefix is documented?

File: streamer/src/tls_certificates.rs [hunk 5]
```diff
@@ -1,58 +1,98 @@
...
+    //        version [0] (1 elem)
+    //          INTEGER  2
+    //        serialNumber CertificateSerialNumber INTEGER (62 bit)
+    //        signature AlgorithmIdentifier SEQUENCE (1 elem)
+    //          algorithm OBJECT IDENTIFIER 1.3.101.112 curveEd25519 (EdDSA 25519 signature algorithm)
+    //        issuer Name SEQUENCE (1 elem)
+    //          RelativeDistinguishedName SET (1 elem)
+    //            AttributeTypeAndValue SEQUENCE (2 elem)
+    //              type AttributeType OBJECT IDENTIFIER 2.5.4.3 commonName (X.520 DN component)
+    //              value AttributeValue [?] UTF8String Solana
```
Comment (lijunwangs) [inline streamer/src/tls_certificates.rs:46]: Can we keep using the old "Solana node" -- it is more appropriate. It is issued by this node -- not Solana networks.

**Review by lijunwangs — COMMENTED** [review]
It took me a while to fully review the changes in this PR and test the changes to verify they are safe and working:
1. I tested against the mainnet -- connections to the existing nodes are okay
2. Tested pubkey spoofing in the certificate (impersonating someone else's pubkey in the X.509 cert) -- the validator is correctly rejecting with BadSignature during TLS1.3 handshaking.
3. I created a manual test tool to compare the cert and key generated using this code and without the code and use openssl to compare the generated DER file. Do not see worthy aberrations other than the issuer name.
The PR while making it safer by eliminating a third party crate does make the code a lot harder to review and maintain -- so the referential links are important.
The code need to be rebased.

**Comment by lijunwangs** [comment]
Superseded by https://github.com/solana-labs/solana/pull/34896. Thanks for your efforts @ripatel-fd

**Pull request closed without merging**